### PR TITLE
基本的なステージング機能の実装

### PR DESCRIPTION
**Pull request by kaitoyama**
### **User description**
- packageのURLをtraPtitechに変更
- 環境変数`STAGING_CHANNEL_ID`で指定されたチャンネルからのリクエストに対して別の処理をする部分を実装

___

### **PR Type**
enhancement, bug_fix

___

### **Description**
- モジュール名を`traPtitech`に変更し、リポジトリの一貫性を確保。
- ステージングチャンネルのリクエストを処理する機能を追加し、特定のチャンネルでの動作をカスタマイズ。
- エラーハンドリングを追加し、ステージング機能の安定性を向上。
- READMEの画像URLを更新し、ドキュメントの正確性を維持。

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>gpt.go</strong><dd><code>モジュール名の更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/gpt/gpt.go

- モジュール名を`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-0c36211fd7b2ccdfa9970448a0522166031ca800870be7276a069ec0e336efb5">+2/-2</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>MessageReceived.go</strong><dd><code>ステージング機能とエラーハンドリングの実装</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/handler/MessageReceived.go

- ステージングチャンネルのリクエストを処理する機能を追加
- エラーハンドリングを追加

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-c558c7342349c5d40c502fd6d5e36e123d4cc10bf9d02cc870c216bf1021ca3d">+24/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OnDirectMessageCreated.go</strong><dd><code>モジュール名の更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/handler/OnDirectMessageCreated.go

- モジュール名を`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-8b448a55f6dafb8b08ea3c5539c56f8605967f39eefddd242dc891473b18598d">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>OnMessageCreated.go</strong><dd><code>モジュール名の更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/handler/OnMessageCreated.go

- モジュール名を`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-8cadfcff54d138e0d3cabaa56524724622e06bcf70cdb3e42e910abb9c70236b">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>handler.go</strong><dd><code>モジュール名の更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

internal/handler/handler.go

- モジュール名を`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-0097c8e59074fdb549fd2466fb349b2814b208df005f9ccf97b14568d51a5f0a">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>main.go</strong><dd><code>モジュール名の更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

main.go

- モジュール名を`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261">+6/-6</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>go.mod</strong><dd><code>モジュール名の更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

go.mod

- モジュール名を`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-33ef32bf6c23acb95f5902d7097b7a1d5128ca061167ec0716715b0b9eeaa5f6">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>README.md</strong><dd><code>画像URLの更新</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

README.md

- 画像URLを`traPtitech`に変更

</details>

  </td>
  <td><a href="https://github.com/traPtitech/BOT_GPT/pull/27/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**: Comment `/help "your question"` on any pull request to receive relevant information

### CODE REVIEW

File: internal/handler/MessageReceived.go [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
 )
 
 func messageReceived(messageText, messagePlainText, channelID string) {
+	if isStaging(channelID) {
+		_, err := bot.PostMessageWithErr(channelID, "ステージング機能が有効です。")
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		return
```
Comment (nokhnaton) [inline internal/handler/MessageReceived.go:19]: ステージングチャンネルだと普通のチャンネルにあるresetやchatの機能は使わないっていうことで大丈夫ですか?

**Review by nokhnaton — COMMENTED** [review]
一点だけコメントしました!
そこの挙動が想定通りなら他は問題なさそうです!

File: internal/handler/MessageReceived.go (hunk 1 above)
Comment (kaitoyama) [reply internal/handler/MessageReceived.go:19]: とりあえず、このPRでは使わなくて、この後のRAGのPoCでは使うコード書いてます

Comment (nokhnaton) [reply internal/handler/MessageReceived.go:19]: なら良さそうです!

**Review by nokhnaton — APPROVED** [review]

**Pull request merged**